src/Lists.php

```diff
@@ -0,0 +1,356 @@
+    }
+
+    /**
+     * Get all lists
+     * @link https://www.klaviyo.com/docs/api/v2/lists#get-lists
+     *
+     * @return bool/mixed
+     */
+    public function getLists() {
+
+        return $this->v2Request( self::LISTS );
+    }
+
+    /**
+     * Get information about a list
+     * @link https://www.klaviyo.com/docs/api/v2/lists#get-list
+     *
+     * @param $listId
+     * 6 digit unique identifier of the list
+     *
+     * @return bool|mixed
+     */
+    public function getListDetails( $listId )
+    {
+        $path = sprintf( '%s/%s', self::LIST, $listId );
+        return $this->v2Request( $path );
+    }
+
+    /**
+     * Update a list's properties
+     * @link https://www.klaviyo.com/docs/api/v2/lists#put-list
+     *
+     * @param $listId
+     * 6 digit unique identifier of the list
+     *
+     * @param $list_name
+     * String to update list name to
+     *
+     * @return bool|mixed
+     */
+    public function updateListDetails( $listId, $list_name )
+    {
+        $params = $this->createRequestBody( array(
+            self::LIST_NAME => $list_name
+        ) );
+
+        $path = sprintf( '%s/%s', self::LIST, $listId );
+
+        return $this->v2Request( $path, $params, self::HTTP_PUT );
+    }
+
+    /**
+     * Delete a list from an account. This is a destructive operation and cannot be undone. It will also remove flow triggers associated with the list.
+     * @link https://www.klaviyo.com/docs/api/v2/lists#delete-list
+     *
+     * @param $listId
+     * 6 digit unique identifier of the list
+     *
+     * @return bool|mixed
+     */
+    public function deleteList( $listId )
+    {
+        $path = sprintf( '%s/%s', self::LIST, $listId );
+        return $this->v2Request( $path, [], self::HTTP_DELETE );
+    }
+
+    /**
+     * Subscribe or re-subscribe profiles to a list. Profiles will be single or double opted into the specified list in accordance with that list’s settings.
+     * @link https://www.klaviyo.com/docs/api/v2/lists#post-subscribe
+     *
+     * @param $listId
+     * 6 digit unique identifier of the list
+     *
+     * @param array $profiles
+     * The profiles that you would like to subscribe. Each object in the list must have either an email or phone number key.
+     * You can also provide additional properties as key-value pairs. If you are a GDPR compliant business, you will need to include $consent in your API call.
+     * $consent is a Klaviyo special property and only accepts the following values: "email", "web", "sms", "directmail", "mobile".
+     * If you are updating consent for a phone number or would like to send an opt-in SMS to the profile (for double opt-in lists), include an sms_consent key in the profile with a value of true or false.
+     *
+     * @return bool|mixed
+     */
+    public function subscribeMembersToList( $listId, array $profiles )
+    {
+        array_walk($profiles, [$this, 'isInstanceOf'], __NAMESPACE__ . '\Model\ProfileModel');
```

Comment:
wondering if we should raise a klaviyo specific exception here if the profiles aren't formatted correctly.

Reply:
regardless, this is very nice!

Reply:
return an exception containg a message w/ how they should be formatted?

Reply:
Yep, that's what I was thinking. Although as I'm looking more closely at [array_walk](https://www.php.net/manual/en/function.array-walk.php) I'm pretty sure it will always return true. In which case this might not be doing what you expect.

Reply:
Yup, looking at it, it just runs the isInstanceOf function of each profile in the array and doesn't care if a profile is not an instance of profile, going to change this a bit and throw an exception if it is not an instance of Profile